package ogp_msg;

message DockerContainerLabel {
    required string name = 1;
    required string value = 2;
}

message DockerContainerPort {
    required int32 private_port = 1;
    required int32 public_port = 2;
    required string type = 3;
}

message DockerContainerInfo {
    required string id = 1;
    required string image = 2;
    repeated string names = 3;
    required string command = 4;
    required int32 created = 5;
    required string status = 6;
    repeated DockerContainerPort ports = 7;
    repeated DockerContainerLabel labels = 8;
}

// 容器本地收集的运行时信息
message DockerRuntimeInfo {
    repeated DockerContainerInfo containers = 1;
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

message DockerTargetContainerPort {
    required int32 private_port = 1;
    required int32 public_port = 2;
    required string type = 3;
}

message DockerTargetVolume {
    required string docker_volume = 1;
    required string host_volume = 2;
}

message DockerTargetDNS {
    required string dns = 1;
    required string address = 2;
}

message DockerTargetContainerInfo {
    required string image = 1;
    required string name = 2;
    repeated DockerTargetContainerPort ports = 3;
    repeated DockerTargetVolume volumes = 4;
    repeated DockerTargetDNS dns = 5;
}

// 容器应该处于的运行时状态信息
message DockerTargetRuntimeInfo {
    repeated DockerTargetContainerInfo containers = 1;
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

message ControllerApplicationVersion {
    required int32 id = 1;
    required string version = 2;
    required string registe_time = 3;
    required string description = 4;
}

message ControllerApplicationInfo {
    required int32 id = 1;
    required string source = 2;
    required string name = 3;
    required string description = 4;
    repeated ControllerApplicationVersion versions = 5;
}

// 获取应用列表信息
message ControllerApplicationsList {
    repeated ControllerApplicationInfo applications = 1;
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

// 消息头
message MsgHeader {
    // rc为0表示成功,否则表示失败。失败原因参见message
    required int32 rc = 1;
    optional string message = 2;
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

// 增加应用请求
message AddApplicationReq {
    required string app_name = 1;
    required string app_version = 2;
    required string app_source = 3;
    required string app_version_desc = 4;
    optional string app_desc = 5;
}

// 增加应用响应
message AddApplicationRes {
    required MsgHeader header = 1;
}
